src/pybind/mgr/orchestrator/module.py

```diff
@@ -823,6 +834,27 @@ def _list_daemons(self,
 
         def ukn(s: Optional[str]) -> str:
             return '<unknown>' if s is None else s
+
+        def sort_by_field(d: DaemonDescription) -> Optional[str]:
+            if sort_by == DaemonFields.name:
+                return d.name()
+            elif sort_by == DaemonFields.host:
+                return d.hostname
+            elif sort_by == DaemonFields.status:
+                return d.status
+            elif sort_by == DaemonFields.refreshed:
+                return d.last_refresh
+            elif sort_by == DaemonFields.age:
+                return d.created
+            elif sort_by == DaemonFields.mem_use:
+                return d.memory_usage
+            elif sort_by == DaemonFields.mem_lim:
+                return d.memory_request
+            elif sort_by == DaemonFields.image:
+                return d.container_image_id
+            else:
+                return None
```

Comment:
some type errors from this if/else block
```
orchestrator/module.py: note: In function "_list_daemons":
orchestrator/module.py:844: error: Incompatible return value type (got "Optional[DaemonDescriptionStatus]", expected "Optional[str]")
orchestrator/module.py:846: error: Incompatible return value type (got "Optional[datetime]", expected "Optional[str]")
orchestrator/module.py:848: error: Incompatible return value type (got "Optional[datetime]", expected "Optional[str]")
orchestrator/module.py:850: error: Incompatible return value type (got "Optional[int]", expected "Optional[str]")
orchestrator/module.py:852: error: Incompatible return value type (got "Optional[int]", expected "Optional[str]")
```

Reply:
Fixed